plugins/woocommerce/tests/e2e-pw/utils/site.js

```diff
@@ -237,6 +237,72 @@ const reset = async ( cKey, cSecret ) => {
 	await deleteAllTaxRates();
 };
 
+/**
+ * Convert Cart and Checkout pages to shortcode.
+ */
+const maybeUseCartCheckoutShortcodes = async ( baseURL, userAgent, admin ) => {
+	/**
+	 * A WordPress page.
+	 * @typedef {Object} WPPage
+	 * @property {number} id
+	 * @property {string} slug
+	 */
+
+	const { request: apiRequest } = require( '@playwright/test' );
+	const { encodeCredentials } = require( './plugin-utils' );
+
+	const basicAuth = encodeCredentials( admin.username, admin.password );
+	const Authorization = `Basic ${ basicAuth }`;
+	const extraHTTPHeaders = {
+		Authorization,
+	};
+	const options = {
+		baseURL,
+		userAgent,
+		extraHTTPHeaders,
+	};
+	const request = await apiRequest.newContext( options );
+
+	// List all pages
+	const response_list = await request.get( '/wp-json/wp/v2/pages', {
+		data: {
+			_fields: [ 'id', 'slug' ],
+		},
+		failOnStatusCode: true,
+	} );
+
+	/**
+	 * @type {WPPage[]}
+	 */
+	const list = await response_list.json();
+
+	// Find the cart and checkout pages
+	const cart = list.find( ( page ) => page.slug === 'cart' );
+	const checkout = list.find( ( page ) => page.slug === 'checkout' );
+
+	// Convert their contents to shortcodes
+	await request.put( `/wp-json/wp/v2/pages/${ cart.id }`, {
+		data: {
+			content: {
+				raw: '<!-- wp:shortcode -->[woocommerce_cart]<!-- /wp:shortcode -->',
+			},
+		},
+		failOnStatusCode: true,
+	} );
+	console.log( 'Cart page converted to shortcode.' );
+
+	await request.put( `/wp-json/wp/v2/pages/${ checkout.id }`, {
+		data: {
+			content: {
+				raw: '<!-- wp:shortcode -->[woocommerce_checkout]<!-- /wp:shortcode -->',
+			},
+		},
+		failOnStatusCode: true,
+	} );
+	console.log( 'Checkout page converted to shortcode.' );
+};
+
```

Comment:
This code looks just the same as the one added in `plugins/woocommerce/tests/api-core-tests/utils/site.js`. I wonder if we should extract this to a generic utils file shared among different kinds of tests such as E2E and API.

Reply:
> I wonder if we should extract this to a generic utils file

I thought about it too. But before we implement that, we'll need to make sure how many of the utility functions are being  shared. If they're too few, then maybe it still makes sense to have this duplication for now.

Reply:
@alopezari if you're fine with it, we can implement the shared utility folder/file between e2e and api tests in a future PR? I'll mark this as resolved for now.